tests/test_new_return_types.py

```diff
@@ -1174,6 +1175,975 @@ def circuit(x):
                     assert isinstance(r, dict)
 
 
+herm = np.diag([1, 2, 3, 4])
+probs_data = [
+    (None, [0]),
+    (None, [0, 1]),
+    (qml.PauliZ(0), None),
+    (qml.Hermitian(herm, wires=[1, 0]), None),
+]
+wires = [2, 3, 4]
+
+
+class TestIntegrationSingleReturn:
```

Comment:
It would be good to create a separate file for these tests. For the interfaces, we have a `test_<mlframework>.py` file for unit tests and `test_<mlframework>_qnode.py` for integration tests with the QNode.

Reply:
Should be good to iterate over shots for this class (same for unit tests now be available in #2).

Reply:
I will rework and separate the tests.